### PR TITLE
cleanup of spline class

### PR DESCRIPTION
**Pull request by tjhei**
- move band_matrix into .cc
- improve documentation
- remove unused variables

### CODE REVIEW

File: include/aspect/utilities.h [hunk 1]
```diff
@@ -109,50 +109,40 @@ namespace aspect
...
       class spline
       {
-        private:
-          std::vector<double> m_x,m_y;           // x,y coordinates of points
-          // interpolation parameters
-          // f(x) = a*(x-x_i)^3 + b*(x-x_i)^2 + c*(x-x_i) + y_i
-          std::vector<double> m_a,m_b,m_c,m_d;
         public:
+          /**
+             * Initialize the spline.
```
Comment (gassmoeller) [inline include/aspect/utilities.h]: The formatting of this comment looks wrong (too many spaces), I do not know why astyle did not catch it. Or is it just my browser?

File: include/aspect/utilities.h [hunk 2]
```diff
@@ -109,50 +109,40 @@ namespace aspect
...
+                          const std::vector<double> &y,
+                          bool cubic_spline = true);
+          /**
+           * Evaluate spline at point @p x.
+           */
           double operator() (double x) const;
+
+        private:
+          /**
+           * x,y coordinates of points
```
Comment (gassmoeller) [inline include/aspect/utilities.h]: Remove the y here

File: source/utilities.cc [hunk 3]
```diff
@@ -183,6 +183,54 @@ namespace aspect
...
     {
+      /**
+       * Band matrix solver for a banded, square matrix of size @p dim. Used by spline interpolation in tk::spline.
+       */
+      class band_matrix
+      {
+        public:
+          band_matrix() {}                              // constructor
+          band_matrix(int dim, int n_u, int n_l);       // constructor
+          void resize(int dim, int n_u, int n_l);       // init with dim,n_u,n_l
```
Comment (gassmoeller) [inline source/utilities.cc]: Would you mind bringing this in our usual style with doxygen formatting? I know it is just copy-paste on your part, but someone should do it, and right now I can force you to it :smile:

File: source/utilities.cc [hunk 4]
```diff
@@ -183,6 +183,54 @@ namespace aspect
...
+          /**
+           * Return the dimension of the matrix
+           */
+          int dim() const;
+
+          /**
+           * number of
```
Comment (gassmoeller) [inline source/utilities.cc]: hmm, number of?

File: source/utilities.cc [hunk 5]
```diff
@@ -183,6 +183,54 @@ namespace aspect
...
+          double &saved_diag(int i);
+          double  saved_diag(int i) const;
+          void lu_decompose();
+          std::vector<double> r_solve(const std::vector<double> &b) const;
+          std::vector<double> l_solve(const std::vector<double> &b) const;
+          std::vector<double> lu_solve(const std::vector<double> &b,
+                                       bool is_lu_decomposed=false);
+        private:
+          std::vector< std::vector<double> > m_upper;  // upper band
+          std::vector< std::vector<double> > m_lower;  // lower band
```
Comment (gassmoeller) [inline source/utilities.cc]: Same here with the comments

**Comment by gassmoeller** [comment]
Thanks for the cleanup :+1: 
Apart from the minor style issues I am fine with the change. 
Feel free to address and merge yourself.

File: include/aspect/utilities.h (hunk 1 above)
Comment (tjhei) [inline include/aspect/utilities.h]: astyle does not care about stuff inside comments, sadly.

**Pull request merged**